lib/Frontend/DiagnosticVerifier.cpp

```diff
@@ -24,132 +24,81 @@
-  case llvm::SourceMgr::DK_Note: return "note";
-  case llvm::SourceMgr::DK_Remark: return "remark";
+  case DiagnosticKind::Error:
+    return "error";
+  case DiagnosticKind::Warning:
+    return "warning";
+  case DiagnosticKind::Note:
+    return "note";
+  case DiagnosticKind::Remark:
+    return "remark";
   }
 
   llvm_unreachable("Unhandled DiagKind in switch.");
 }
 
-
-
-namespace {
-  /// This class implements support for -verify mode in the compiler.  It
-  /// buffers up diagnostics produced during compilation, then checks them
-  /// against expected-error markers in the source file.
-  class DiagnosticVerifier {
-    SourceManager &SM;
-    std::vector<llvm::SMDiagnostic> CapturedDiagnostics;
-  public:
-    explicit DiagnosticVerifier(SourceManager &SM) : SM(SM) {}
-
-    void addDiagnostic(const llvm::SMDiagnostic &Diag) {
-      CapturedDiagnostics.push_back(Diag);
-    }
-
-    /// Result of verifying a file.
-    struct Result {
-      /// Were there any errors? All of the following are considered errors:
-      /// - Expected diagnostics that were not present
-      /// - Unexpected diagnostics that were present
-      /// - Errors in the definition of expected diagnostics
-      bool HadError;
-      bool HadUnexpectedDiag;
-    };
-
-    /// verifyFile - After the file has been processed, check to see if we
-    /// got all of the expected diagnostics and check to see if there were any
-    /// unexpected ones.
-    Result verifyFile(unsigned BufferID, bool autoApplyFixes);
-
-    /// diagnostics for '<unknown>:0' should be considered as unexpected.
-    bool verifyUnknown();
-
-    void printRemainingDiagnostics() const;
-
-    /// If there are any -verify errors (e.g. differences between expectations
-    /// and actual diagnostics produced), apply fixits to the original source
-    /// file and drop it back in place.
-    void autoApplyFixes(unsigned BufferID,
-                        ArrayRef<llvm::SMDiagnostic> diagnostics);
-    
-  private:
-    std::vector<llvm::SMDiagnostic>::iterator
-    findDiagnostic(const ExpectedDiagnosticInfo &Expected,
-                   StringRef BufferName);
-
-  };
-} // end anonymous namespace
-
-
-
-
 /// If we find the specified diagnostic in the list, return it.
 /// Otherwise return CapturedDiagnostics.end().
-std::vector<llvm::SMDiagnostic>::iterator
-DiagnosticVerifier::findDiagnostic(const ExpectedDiagnosticInfo &Expected,
-                                   StringRef BufferName) {
+static std::vector<CapturedDiagnosticInfo>::iterator
+findDiagnostic(std::vector<CapturedDiagnosticInfo> &CapturedDiagnostics,
+               const ExpectedDiagnosticInfo &Expected, StringRef BufferName) {
   for (auto I = CapturedDiagnostics.begin(), E = CapturedDiagnostics.end();
        I != E; ++I) {
     // Verify the file and line of the diagnostic.
-    if (I->getLineNo() != (int)Expected.LineNo ||
-        I->getFilename() != BufferName)
+    if (I->Loc.isInvalid())
```

Comment:
Why this extra condition?

Reply:
This was left over from a previous refactor, I removed it